homeassistant/components/azure_data_explorer/__init__.py

```diff
@@ -0,0 +1,212 @@
+"""The Azure Data Explorer integration."""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+import json
+import logging
+
+from azure.kusto.data.exceptions import KustoAuthenticationError, KustoServiceError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import MATCH_ALL
+from homeassistant.core import Event, HomeAssistant, State
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.entityfilter import FILTER_SCHEMA
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.json import JSONEncoder
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util.dt import utcnow
+
+from .client import AzureDataExplorerClient
+from .const import (
+    CONF_APP_REG_SECRET,
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DATA_FILTER,
+    DATA_HUB,
```

Comment:
```suggestion
```

Reply:
By using all these changes, the component will no longer be able to be un-loaded. I have tried many different ways, but I cannot get this approche to work

Reply:
I can see that the name DATA_HUB need to change to something else like ADX

Reply:
We can change it also in a follow up PR, so this big one gets merged :)